src/chat/ChatScreen.js

```diff
@@ -125,20 +128,41 @@ export default function ChatScreen(props: Props) {
   const sayNoMessages = haveNoMessages && !isFetching;
   const showComposeBox = canSendToNarrow(narrow) && !showMessagePlaceholders;
 
+  const auth = useSelector(getAuth);
   const dispatch = useDispatch();
   const caughtUp = useSelector(state => getCaughtUpForNarrow(state, narrow));
   const _ = useContext(TranslationContext);
 
-  const sendMessage = useCallback(
+  const sendCallback = useCallback(
     (message: string, destinationNarrow: Narrow) => {
-      if (!caughtUp.newer) {
-        showErrorAlert(_('Failed to send message'));
-        return;
-      }
+      if (editMessage) {
+        const content = editMessage.content !== message ? message : undefined;
+        const subject = caseNarrowDefault(
```

Comment:
Why the change to start using `caseNarrowDefault`? I think it belongs in a different commit; this one is described as a simple "switch from A to B" commit.

Ah, looking again: possibly this was a small rebase error? 🙂

Reply:
Hmmm, I think I had this in there the whole time, and I agree that the commit doesn't describe that. The reason that it's needed is that the previous code could look directly at the `state.topic`, but the callback doesn't have access to the state.

I've added a commit to make that change in the original code.